https://github.com/dials/cbflib/pull/68
--- a/src/cbf_simple.c
+++ b/src/cbf_simple.c
@@ -669,7 +669,7 @@
                     
                     match = 0;
                     
-                    for(crow = 0; crow < count+1; crow++) {
+                    for(crow = 0; crow < count; crow++) {
                         
                         if (!cbf_cistrcmp(xscan_id,scanids[crow])) {
                             
